build-system/tasks/performance/measure-documents.js

```diff
@@ -118,6 +181,95 @@ const readMetrics = (page) =>
     };
   });
 
+/**
+ * Set up defaults handlers
+ * @param {!Array<function>} handlersList
+ */
+function setUpDefaultHandlers(handlersList) {
+  // Always rewrite vendor json requests to pull locally.
+  // That way we are not including the time difference between
+  // fetching a local file vs network request to CDN.
+  handlersList.push(analyticsVendorRewriteHandler);
+}
+
+/**
+ * Set up appropriate handlers
+ * @param {!Array<function>} handlersList
+ * @param {?Object} handlerOptions
+ */
+function setUpAddionalHandlers(handlersList, handlerOptions) {
+  // Setting up timing and adding specified handler
+  if (handlerOptions.handlerName === 'analyticsHandler') {
```

Comment:
Correct me if this wrong. 
All urls within the same `handlers` config share the same `handlerOptions` config. (we're not creating new instances of the `handlerOptions` object within `load-config.js` file). I'm afraid rewriting the `handlerOptions` object would affect other instances that fall within the same handler. Have you tried the test with multiple urls.

Reply:
Yes I have tried it with multiple urls; within `measureDocument()`, we create `handlerOptionsForUrl` by creating a shallow copy of the original `handlerOptions` config. This works fine for storing the `startTime` and `endTime`, but would cause some bugs if someone were to mutate objects within the `handlerOptions`. Instead of shallow copy, should I create a deep copy?